### PR TITLE
[osx] - fix timebase of vblankhandler

### PR DESCRIPTION
**Pull request by Memphiz**
as spotted by @FernetMenta - thx

### CODE REVIEW

**Comment by Memphiz** [comment]
jenkins build this please

**Comment by Memphiz** [comment]
jenkins build this please (http error 502 during trigger)

**Comment by Memphiz** [comment]
@FernetMenta could you test this PR? I moved the init/deinit call into the vsynchandler thread and implemented the same approach like with GLX. Not quiet sure if its bullet proof though because there could be a display reset in the very moment where the vsync run is called imo.

File: xbmc/video/videosync/VideoSyncOsx.cpp [hunk 1]
```diff
@@ -73,18 +81,27 @@ float CVideoSyncOsx::GetFps()
...
+}
+
 void CVideoSyncOsx::OnResetDevice()
 {
-  m_abort = true;
+  m_displayReset = true;
 }
 
-void CVideoSyncOsx::VblankHandler(int64_t nowtime)
+void CVideoSyncOsx::VblankHandler(int64_t nowtime, int32_t timebase)
```
Comment (FernetMenta) [inline xbmc/video/videosync/VideoSyncOsx.cpp]: UpdateClock is still called with nowtime which is not the same timebase as videorefclock

Comment (Memphiz) [inline xbmc/video/videosync/VideoSyncOsx.cpp]: @FernetMenta should i call updateclock with CurrentHostCounter instead of nowtime?

Comment (FernetMenta) [inline xbmc/video/videosync/VideoSyncOsx.cpp]: yes, ideally you take the snapshot of CurrentHostCounter at entry of function.

**Comment by FernetMenta** [comment]
Seems this links to the wrong display when having multiple monitors. VblankHandler is called every 16ms though video runs on 2nd 24 Hz display.

**Comment by Memphiz** [comment]
Updated with all fixes i found. I would still like to have this in Isengard. It was tested by me during the whole weekend, fernet tested as well and at least 3 users in the forum posted that this fixes the sync to display when using multiple monitors. This needs to go in after #7149 (will rebase once it has been merged). Well not sure who i am to ask - maybe @FernetMenta (again sorry) for the final saying

**Comment by FernetMenta** [comment]
This PR fixes a whole bunch of issues including crashes. I can't think of any good reasons why it shouldn't go in for Isengard.

**Comment by fritsch** [comment]
jenkins build this please

**Comment by fritsch** [comment]
@Memphiz @FernetMenta if we need it, nothing speaks against pushing it now, right?

**Comment by Memphiz** [comment]
let me rebase first

**Comment by Memphiz** [comment]
thx @FernetMenta for spending the weekend with me haha ;)

File: xbmc/windowing/osx/WinSystemOSX.mm [hunk 2]
```diff
@@ -797,6 +814,7 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   m_nWidth = newWidth;
   m_nHeight = newHeight;
   m_glContext = context;
+  g_graphicsContext.SetFPS(m_refreshRate);
```
Comment (FernetMenta) [inline xbmc/windowing/osx/WinSystemOSX.mm]: @Memphiz this is tool late and leaves GraphicsContext in an inconsistent state. If you set an override to fps, you have to do this in before you fire OnResetDisplay event. 
Now videoSyncOsx receives the ResetDisplay event and gets incorrect FPS from graphicsContext.

Comment (Memphiz) [inline xbmc/windowing/osx/WinSystemOSX.mm]: i have put a fix in my https://github.com/Memphiz/xbmc/commits/vtb_choppy_bp branch and jenkins does a build right now. I let the users test then.

It contains - the fix for the window origin + your idispresource commit + setting the graphicscontext fps before calling OnResetDisplay callbacks and i've put the wait before the flushbuffers...

Comment (Memphiz) [inline xbmc/windowing/osx/WinSystemOSX.mm]: @FernetMenta does this test build correct what you spotted?

http://mirrors.kodi.tv/test-builds/osx/x86_64/kodi-20170103-e6d544f-vtb_choppy_bp-x86_64.dmg

**Pull request merged**